[2025-10-02T16:36:18.782+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_pipeline_scd2.load_to_dw_scd2 manual__2025-10-02T16:34:01.099306+00:00 [queued]>
[2025-10-02T16:36:18.796+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_pipeline_scd2.load_to_dw_scd2 manual__2025-10-02T16:34:01.099306+00:00 [queued]>
[2025-10-02T16:36:18.798+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2025-10-02T16:36:18.836+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): load_to_dw_scd2> on 2025-10-02 16:34:01.099306+00:00
[2025-10-02T16:36:18.857+0000] {standard_task_runner.py:60} INFO - Started process 591 to run task
[2025-10-02T16:36:18.870+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'etl_pipeline_scd2', 'load_to_dw_scd2', 'manual__2025-10-02T16:34:01.099306+00:00', '--job-id', '59', '--raw', '--subdir', 'DAGS_FOLDER/process_scripts.py', '--cfg-path', '/tmp/tmplsh_lkmt']
[2025-10-02T16:36:18.876+0000] {standard_task_runner.py:88} INFO - Job 59: Subtask load_to_dw_scd2
[2025-10-02T16:36:18.914+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.11/site-packages/***/settings.py:194 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-10-02T16:36:19.010+0000] {task_command.py:423} INFO - Running <TaskInstance: etl_pipeline_scd2.load_to_dw_scd2 manual__2025-10-02T16:34:01.099306+00:00 [running]> on host f327b3c387fa
[2025-10-02T16:36:19.173+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='etl_pipeline_scd2' AIRFLOW_CTX_TASK_ID='load_to_dw_scd2' AIRFLOW_CTX_EXECUTION_DATE='2025-10-02T16:34:01.099306+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-10-02T16:34:01.099306+00:00'
[2025-10-02T16:36:21.668+0000] {scd2_loader.py:121} WARNING - 1 lignes ignorées dans /opt/***/data/out/Country_code.xlsx à cause de clés business manquantes
[2025-10-02T16:36:21.744+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'NG'})
[2025-10-02T16:36:21.749+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'YT'})
[2025-10-02T16:36:21.752+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'ER'})
[2025-10-02T16:36:21.757+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'ET'})
[2025-10-02T16:36:21.763+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'MG'})
[2025-10-02T16:36:21.769+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'MA'})
[2025-10-02T16:36:21.775+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'MU'})
[2025-10-02T16:36:21.780+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'RE'})
[2025-10-02T16:36:21.784+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'TN'})
[2025-10-02T16:36:21.788+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'TG'})
[2025-10-02T16:36:21.793+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'TD'})
[2025-10-02T16:36:21.797+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'SN'})
[2025-10-02T16:36:21.801+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'ML'})
[2025-10-02T16:36:21.805+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'GH'})
[2025-10-02T16:36:21.809+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'GN'})
[2025-10-02T16:36:21.813+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'BW'})
[2025-10-02T16:36:21.815+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'ZA'})
[2025-10-02T16:36:21.820+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'SZ'})
[2025-10-02T16:36:21.823+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'MW'})
[2025-10-02T16:36:21.826+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'AO'})
[2025-10-02T16:36:21.829+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'MZ'})
[2025-10-02T16:36:21.833+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'ZW'})
[2025-10-02T16:36:21.836+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'ZM'})
[2025-10-02T16:36:21.840+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'CM'})
[2025-10-02T16:36:21.843+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'CD'})
[2025-10-02T16:36:21.846+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'GA'})
[2025-10-02T16:36:21.849+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'KE'})
[2025-10-02T16:36:21.852+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'UG'})
[2025-10-02T16:36:21.854+0000] {scd2_loader.py:146} INFO - [INSERT] Nouvelle ligne ajoutée dans `dim_pays` ({'country_code': 'TZ'})
[2025-10-02T16:36:21.871+0000] {scd2_loader.py:169} INFO - [LOAD] /opt/***/data/out/Country_code.xlsx → dim_pays chargé avec SCD2
[2025-10-02T16:36:22.054+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/process_scripts.py", line 28, in run_scd2_loader
    load_all_out_files_to_dw()
  File "/opt/airflow/dags/scd2_loader.py", line 119, in load_all_out_files_to_dw
    valid_rows = df.dropna(subset=business_keys)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/pandas/core/frame.py", line 6418, in dropna
    raise KeyError(np.array(subset)[check].tolist())
KeyError: ['Cost Center']
[2025-10-02T16:36:22.092+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=etl_pipeline_scd2, task_id=load_to_dw_scd2, execution_date=20251002T163401, start_date=20251002T163618, end_date=20251002T163622
[2025-10-02T16:36:22.123+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 59 for task load_to_dw_scd2 (['Cost Center']; 591)
[2025-10-02T16:36:22.175+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-10-02T16:36:22.195+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
